packages/core/src/transports/multiplexed.ts

```diff
@@ -0,0 +1,84 @@
+import type {
+  BaseTransportOptions,
+  Envelope,
+  Event,
+  EventItem,
+  Transport,
+  TransportMakeRequestResponse,
+} from '@sentry/types';
+import { dsnFromString, forEachEnvelopeItem } from '@sentry/utils';
+
+import { getEnvelopeEndpointWithUrlEncodedAuth } from '../api';
+
+interface MatchParam {
+  // The envelope to be sent
+  envelope: Envelope;
+  // A function that returns an event from the envelope if one exists
+  getEvent(): Event | undefined;
```

Comment:
```suggestion
  getEvent(envelope: Envelope): Event | undefined;
```

Shouldn't this signature look like this?

Reply:
The `getEvent` function passed to the matcher already passes the envelope. This saves consumers of this API from needing to import another function.

Reply:
ahh I see, sorry, didn't understand this correctly. makes sense 👍